### PR TITLE
Add spag request

### PR DESCRIPTION
**Pull request by pglass**
- [x] Respect the `-H <header>` flag for overriding headers
- [x] Respect the `-d <data>` flag for overriding the body
- [x] Print out paths relative to the current dir on a `spag request list`
- [x] Python tests should pass on `nosetests -sv tests/test_spag.py -m "request"`

Probably templates will happen in a separate pull request

### CODE REVIEW

File: src/spag/main.rs [hunk 1]
```diff
@@ -118,14 +157,83 @@ fn spag_history(args: &Args) {
...
+    let filenames = try_error!(file::walk_dir(&dir));
+    let mut yaml_files: Vec<&PathBuf> = filenames.iter()
+        .filter(|p| p.to_str().unwrap().ends_with(".yml"))
+        .collect();
+    yaml_files.sort();
+
+    let current_dir = try_error!(std::env::current_dir());
+    for file in yaml_files.iter() {
+        // relative_from() is unstable
+        println!("{}", file.relative_from(&current_dir).unwrap().to_str().unwrap());
```
Comment (Squab) [inline src/spag/main.rs]: This didn't seem to work, but I believe
` println!("{}", file.to_str().unwrap());`
did.

Comment (pglass) [inline src/spag/main.rs]: What version of Rust do you have? I have:

    $ rustc --version
    rustc 1.2.0-nightly (a9515698f 2015-06-20)

And this definitely works for me. Or `relative_from` may work on Linux and not OS X which would be super dumb.

Comment (Squab) [inline src/spag/main.rs]: I'm on
```
$ rustc --version
rustc 1.2.0-nightly (c6b148337 2015-06-12)
```

Let me upgrade and see what happens.

**Comment by Squab** [comment]
All good on updated rust. Mergy :+1: :100: :bomb:

**Pull request merged**